lib/common/socket/evloop.c.h

```diff
@@ -19,11 +19,58 @@
  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  * IN THE SOFTWARE.
  */
+
+#include <stdlib.h>
+#ifdef _WIN32
+
+struct iovec {
+   char *buf;
+   unsigned long len;
+};
+
+static ssize_t
+writev(SOCKET fd, const struct iovec *iov, int iovcnt) {
+    DWORD sent = 0;
+    h2o_iovec_t* psend = (h2o_iovec_t*) iov;
+    LPWSABUF pbuf = alloca(sizeof(WSABUF) * iovcnt);
+    int i;
+    for (i = 0; i < iovcnt; ++i) {
+        pbuf[i].len = psend[i].len;
+        pbuf[i].buf = psend[i].base;
+        sent += pbuf[i].len;
+    }
+    if (WSASend(fd, pbuf, iovcnt, NULL, 0, NULL, NULL) == 0)
+        return (ssize_t) sent;
+    return -1;
+}
+
+# undef  socket_error
+# undef  read_socket
+# undef  close_socket
+# undef  EINTR
+# define EINTR        WSAEINTR
+# undef  EWOULDBLOCK
+# define EWOULDBLOCK  WSAEWOULDBLOCK
+# undef  EAGAIN
+# define EAGAIN       WSAEWOULDBLOCK
+# undef  EINPROGRESS
+# define EINPROGRESS  WSAEINPROGRESS
+
+# define socket_error WSAGetLastError()
+# define read_socket(f, d, s) recv(f, d, s, 0)
+# define close_socket(f) closesocket(f)
```

Comment:
Is there any reason to prefix (or suffix) the names with `socket`?

IMO it would be preferable to continue using names like `read`, `errno`, `close`.